imgcreate/live.py

```diff
@@ -320,10 +320,23 @@ def __create_iso(self, isodir):
         iso = self._outdir + "/" + self.name + ".iso"
 
         args = ["xorrisofs",
-                "-joliet", "-rational-rock",
-                "-hide-rr-moved",
-                "-volid", self.fslabel,
-                "-output", iso]
+                "-output", iso,
+                "-isohybrid-mbr", "/usr/share/syslinux/isohdpfx.bin",
+                "-eltorito-boot", "isolinux/isolinux.bin",
+                "-eltorito-catalog", "isolinux/boot.cat",
+                "-boot-load-size", "4",
+                "-boot-info-table", "-no-emul-boot"]
+
+        def iterate_files_recursively(directory="."):
+            for entry in os.scandir(directory):
+                if entry.is_dir():
+                    for file in iterate_files_recursively(entry.path):
+                        yield file
+                else:
+                    yield entry.stat().st_size
+
+        if max(iterate_files_recursively(isodir)) >= 4 * 1024**3:
+            [args.insert(1, i) for i in ['3', '-iso-level']]
```

Comment:
That's kind of an odd way to add the argument? Why did you do it that way?

Reply:
I like the concise form of a [list comprehension](http://www.siafoo.net/article/52#id18). Reversing the insertion list makes it easy to match the order desired.

Reply:
Sorry, I force pushed a more traditional approach before seeing that you may have offered a change (that is now outdated).

Reply:
Nope, I hadn't added any other comments. I'm a fan of list comprehension as well, but that line is *really* hard to understand on first or second look :) The updated version is much easier to read.